### PR TITLE
docs: purpose

### PR DESCRIPTION
**Pull request by nickbytes**
WIP – what else do you think I could add that would be helpful @jyng ? cc @g-a-v-i-n

### CODE REVIEW

File: PURPOSE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+Projects are moving targets with changing requirements, so sometimes these components would need updates or tweaks. A developer might update their indigo components or classes, causing their version to be out of sync with the `master` copy, and also out of sync with all other developers.
+
+New, useful components that were created through project work were tough to integrate into the indigo project. If updates were made to indigo, one would have to update all the applications using indigo by hand, which is error prone and slow. There's also no mechanism to extend components in a sane manner that everyone can understand.
+
+## Updates
+
+1. Separate the projects – build tooling, publishing, and project structure are pretty different for static css, react component libraries, and design tokens. We have separated the projects. The common denominator is that they are all now part of the Javascript ecosystem (more below).
```
Comment (jyng) [inline PURPOSE.md]: We should also talk how it all works together. ie; all variables live inside indigo-tokens which is then read by indigo-react, etc

File: PURPOSE.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+3. Make maintainence easier – the react library comes with an example site (create-react-app) where your components are available _as you develop_. You can use this as a playground for new ideas, a documentation site, or a learning resource for other developers.
+
+4. CSS-in-JS – These components are being used in mostly React applications because that's the `planet` we live on. You can use these components on the web, React Native projects, or Electron apps. Some urbit projects already use atomic css (which is great). Styled-components + styled-system is like the CSS-in-JS version of atomic css. Atomic CSS in a component library can get a little weird. Each component, which was supposed to be isolated, somewhat relies on a global css import. Even if we want to use 1 component from the library, we still need to import all of the CSS (just because this is easiest in a sass/css pipeline). With CSS-in-JS solutions we eliminate global css import, have easier built-in methods to extends components, and can still leverage theme tokens. We can also import only the components/styles that we need. We are also able to slowly transition away from the old components.
```
Comment (jyng) [inline PURPOSE.md]: It'd be nice to just have some bullet points on what benefits css-in-js bring us

Comment (nickbytes) [reply PURPOSE.md]: Updated, let me know what you think of new version.

File: PURPOSE.md (hunk 1 above)
Comment (nickbytes) [reply PURPOSE.md]: Updated, let me know what you think of new version.

**Pull request merged**